### ISSUE TITLE
Bug: Accessibility failures with DatePicker and TimePicker components

### ISSUE THREAD

**Issue by ryshah** [carbon-design-system/carbon-components-react#93]
When trying to integrate DatePicker and TimePicker components, we observed following accessibility issues with July 2016 rules. 

**Error**: Each form control must have associated label.
**Code Snippets:**
- `<select class=\"bx--select-input\">`
- `<input class=\"bx--date-picker__input flatpickr-input\" pattern=\"d{1,2}/d{1,2}/d{4}\" placeholder=\"mm/dd/yyyy\" type=\"text\">`
- `<input class=\"bx--time-picker__input-field\" maxlength=\"5\" placeholder=\"hh:mm\" pattern=\"(1[012]|[1-9]):[0-5][0-9](\\s)?(?i)\" type=\"text\">`
